routing/localchans/manager.go

```diff
@@ -156,11 +167,28 @@ func (r *Manager) UpdatePolicy(newSchema routing.ChannelPolicy,
 				))
 
 		default:
-			failedUpdates = append(failedUpdates,
-				makeFailureItem(chanPoint,
-					lnrpc.UpdateFailure_UPDATE_FAILURE_UNKNOWN,
-					"could not update policies",
-				))
+			// If the edge was not found, but the channel is found,
```

Comment:
I don't think we should touch the `default` case here - there seem to be other expected errors other than edge not found. We can instead add a new `case errors.Is(err, channeldb.ErrEdgeNotFound)` and perform the edge creation.

It'd be better if we could also distinguish the case where the edge cannot be found vs the edge policy cannot be found.

Reply:
agree, so we basically need to check failedUpdates for the error which actually occurred when trying to update the policy ?

Reply:
Looking at this now, this would require a rewrite of the function. The thing is `ForAllOutgoingChannels` doesn't touch the channel at all, because it's not found in the graph. Therefore it is still in `unprocessedChans`. Then you fetch the channel. Fetching the channel doesn't fail, and the channel is not pending, so we're sure this is a good channel without a corresponding edge?

Reply:
Done in the last push

Reply:
maybe what yy meant is, that while traversing through the graph and all our channels there might be a slight change that we fail fetching the policy for the specific edge, which might lead us to exit early and never hit this default case ? But normally we take care of adding an unknown policy as soon as we add and `edgeInfo` to the graph db, so I think you are right.